### PR TITLE
Updates the Response of Resources API's by adding array of Categories 

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
- This patch updates the design of Resources by adding
  an array of Categories for the following API's

     - /resource/{catalog}/{kind}/{name}/{version}
     - /resource/version/{versionId}
     - /v1/resource/{catalog}/{kind}/{name}/{version}
     - /v1/resource/version/{versionId}

- Updates the resources tests which adds category array in the
  response of CatalogKindNameVersion and ResourceBYVersionID

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/pkg/shared/resource/resource.go [hunk 1]
```diff
@@ -215,6 +215,7 @@ func withResourceVersionDetails(db *gorm.DB) *gorm.DB {
 	return db.
 		Preload("Resource").
 		Preload("Resource.Catalog").
+		Preload("Resource.Category").
```
Comment (pratap0007) [inline api/pkg/shared/resource/resource.go:218]: WDYT: Ordering of resource's categories as well like resource's tags

Comment (sm43) [reply api/pkg/shared/resource/resource.go]: +1 Can we add this?

Comment (PuneetPunamiya) [reply api/pkg/shared/resource/resource.go]: Will add it

**Pull request merged**